packages/ag-charts-community/src/chart/series/seriesStateManager.ts

```diff
@@ -21,39 +21,40 @@ type SeriesLike = SeriesIdLike & {
     visible: boolean;
 };
 
+type SeriesGroupingEntry = {
+    grouping: SeriesGrouping;
+    visible: boolean;
+};
+
 export class SeriesStateManager {
-    private readonly groups: {
-        [type: string]: {
-            [id: string]: {
-                grouping: SeriesGrouping;
-                visible: boolean;
-            };
-        };
-    } = {};
+    private readonly groups: Map<string, Map<string, SeriesGroupingEntry>> = new Map();
 
     public registerSeries({ internalId, seriesGrouping, visible, type }: SeriesLike) {
         if (!seriesGrouping) return;
 
-        this.groups[type] ??= {};
-        this.groups[type][internalId] = { grouping: seriesGrouping, visible };
+        if (!this.groups.has(type)) {
+            this.groups.set(type, new Map());
+        }
+        this.groups.get(type)?.set(internalId, { grouping: seriesGrouping, visible });
```

Comment:
You can avoid some additional lookups here

```suggestion
        let group = this.groups.get(type);
        if (group == null) {
            group = new Map();
            this.groups.set(type, group);
        }
        group.set(internalId, { grouping: seriesGrouping, visible });
```